subworkflows/local/bgc.nf

```diff
@@ -0,0 +1,100 @@
+        UNTAR_DETECTION ( [ [], ch_detection_for_antismash ] )
+        UNTAR_MODULES ( [ [], ch_modules_for_antismash ] )
+
+        ANTISMASH_ANTISMASHLITEDOWNLOADDATABASES ( UNTAR_CSS.out.untar.map{ it[1] }, UNTAR_DETECTION.out.untar.map{ it[1] }, UNTAR_MODULES.out.untar.map{ it[1] } )
+        ch_versions = ch_versions.mix(ANTISMASH_ANTISMASHLITEDOWNLOADDATABASES.out.versions)
+        ch_antismash_directory = ANTISMASH_ANTISMASHLITEDOWNLOADDATABASES.out.antismash_dir
+
+    } else if ( !params.bgc_antismash_databases && params.bgc_antismash_directory ) {
+
+        // Supply databases
+        ch_css_for_antismash = "https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/antismash/css.tar.gz"
+        ch_detection_for_antismash = "https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/antismash/detection.tar.gz"
+        ch_modules_for_antismash = "https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/antismash/modules.tar.gz"
+
+        UNTAR_CSS ( [ [], ch_css_for_antismash ] )
+        UNTAR_DETECTION ( [ [], ch_detection_for_antismash ] )
+        UNTAR_MODULES ( [ [], ch_modules_for_antismash ] )
+
+        ANTISMASH_ANTISMASHLITEDOWNLOADDATABASES ( UNTAR_CSS.out.untar.map{ it[1] }, UNTAR_DETECTION.out.untar.map{ it[1] }, UNTAR_MODULES.out.untar.map{ it[1] } )
+        ch_versions = ch_versions.mix(ANTISMASH_ANTISMASHLITEDOWNLOADDATABASES.out.versions)
+        ch_antismash_databases = ANTISMASH_ANTISMASHLITEDOWNLOADDATABASES.out.database
+
+        // Supply antiSMASH directory
+        ch_antismash_directory = Channel
+            .fromPath( params.bgc_antismash_directory )
+            .first()
+
+    } else if ( !params.bgc_antismash_databases && !params.bgc_antismash_directory ) {
+
+        // Supply databases
+        ch_css_for_antismash = "https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/antismash/css.tar.gz"
+        ch_detection_for_antismash = "https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/antismash/detection.tar.gz"
+        ch_modules_for_antismash = "https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/antismash/modules.tar.gz"
+
+        UNTAR_CSS ( [ [], ch_css_for_antismash ] )
+        UNTAR_DETECTION ( [ [], ch_detection_for_antismash ] )
+        UNTAR_MODULES ( [ [], ch_modules_for_antismash ] )
+
+        ANTISMASH_ANTISMASHLITEDOWNLOADDATABASES ( UNTAR_CSS.out.untar.map{ it[1] }, UNTAR_DETECTION.out.untar.map{ it[1] }, UNTAR_MODULES.out.untar.map{ it[1] } )
+        ch_versions = ch_versions.mix(ANTISMASH_ANTISMASHLITEDOWNLOADDATABASES.out.versions)
+        ch_antismash_databases = ANTISMASH_ANTISMASHLITEDOWNLOADDATABASES.out.database
+
+        // Supply antiSMASH directory
+        ch_antismash_directory = ANTISMASH_ANTISMASHLITEDOWNLOADDATABASES.out.antismash_dir
+
+    }
+
+    ANTISMASH_ANTISMASHLITE ( fna, ch_antismash_databases, ch_antismash_directory, gff.map{ it[1] } )
```

Comment:
Are you sure the `fna` and `gff` are linked here? I assume ANTISMASH wants both of the same contig/sample right?

Remember Nextflow allows asyncronous emission of objects into a channel, i.e.

You cannot assume the output channels from prokka will be ordered like so

```
Channel A = [ Sample_A, Sample_B, Sample_C ] 
Channel B = [ Sample_A, Sample_B, Sample_C ]
```

It could be like so:

```
Channel A = [ Sample_A, Sample_B, Sample_C ] 
Channel B = [ Sample_B, Sample_A,  Sample_C ]
```

So the first element of Channel A and Channel B may not necessarily correspond to the same input sample.

It is likely safer to `.mix` the fna and gff channels, then `groupTuple` to have `[meta, fna, gff ]`, then you can use `.multiMap` to supply as separate 'channels' to the `ANTISMASHLITE` module but stil have them liinked together.